src/typings/callbacks.d.ts

```diff
@@ -0,0 +1 @@
+export type Callback<Te, Tv> = (err: Te | null | undefined, val?: Tv) => void;
```

Comment:
instead  of `| undefined` you can do `err?`

Reply:
Sorry, I'm not sure I understand. Could you please clarify?

Reply:
Your declaration could be:
`export type Callback<V> = (err?: Error | null, val?: V) => void;`

Reply:
Is that not different? Doesn't `err?` make it an optional parameter, so I could pass in `Callback(Tv)` and ignore `err` altogether. I thought I was making the `err` param required, but it can be `null|undefined`, as long as it's passed in to the callback.

Reply:
It make it optional but you will never be able to ignore it if you want to pass the val.
Also ignore === undefined
That is why `err?` would work

Reply:
OK it makes sense. I found a good explanation that might clarify a bit:
>An optional property can be omitted from an object literal or assigned the value undefined. An | undefined property can be assigned undefined but cannot be omitted from an object literal. | null is the same but with null instead of undefined.
>
>In other words, optional props are not identical to unions, because only the optional marker allows the property to be omitted from object literals.

In this case, because the parameter is first, it must be declared in order to pass a callback function as you mentioned. So the way it's been implemented and what you are suggesting seem to be the exact same thing. Except, I feel like the syntax already used (without the `?`) is more explicit in that the parameter `err` must be passed regardless, whether or not it's value evaluates to `null`, `undefined`, or a <Te>value`

Reply:
And i just read from the handbook:
> With --strictNullChecks, an optional parameter automatically adds | undefined

So, again confirms both solutions do the same :)